### PR TITLE
Expose accounts endpoints

### PR DESCRIPTION
**Pull request by miguelnietoa**
## Description
This PR exposes the CRUD actions for `accounts` together with the recover endpoint.

Fixes #27 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  use MintacoinWeb, :controller
+
+  alias Mintacoin.Accounts
+  alias Mintacoin.Account
+
+  @type params :: map()
+  @type conn :: Plug.Conn.t()
+  @type error :: :not_found | :bad_request | Ecto.Changeset.t()
+
+  @allowed_params ["email", "name"]
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:11]: @miguelnietoa please, use a sigil instead

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  action_fallback MintacoinWeb.FallbackController
+
+  @spec create(conn :: conn(), params :: params()) :: {:error, error()} | conn()
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:15]: @miguelnietoa
```elixir
.
.
@spec create(conn :: conn(), params :: params()) :: conn() | {:error, error()}
```

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    else
+      {:error, :not_found} -> {:error, :not_found}
+      _error -> {:error, :bad_request}
+    end
+  end
+
+  @spec delete(conn :: conn(), params :: params()) :: {:error, error()} | conn()
+  def delete(conn, %{"address" => address}) do
+    case Accounts.delete(address) do
+      {:ok, _account} -> send_resp(conn, :no_content, "")
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:63]: @miguelnietoa why return an empty body response?

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  @spec update(conn :: conn(), params :: params()) :: {:error, error()} | conn()
+  def update(conn, %{"address" => address} = params) do
+    with allowed_params when allowed_params != %{} <- fetch_allowed_params(params),
+         {:ok, account} <- Accounts.update(address, allowed_params) do
+      conn
+      |> put_status(:ok)
+      |> render("account.json", account: account)
+    else
+      {:error, :not_found} -> {:error, :not_found}
+      _error -> {:error, :bad_request}
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:56]: @miguelnietoa this seems to be a typical pattern across actions. Maybe using a common function with pattern matching may simplify, what do you think?

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    end
+  end
+
+  @spec fetch_allowed_params(params :: params()) :: params()
+  defp fetch_allowed_params(params) do
+    params
+    |> Map.take(@allowed_params)
+    |> Enum.reduce(%{}, fn {key, value}, acc ->
+      key
+      |> String.to_existing_atom()
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:84]: @miguelnietoa Atoms are not garbage collected. Think about what will happen with the system memory if someone performs a request including a large set of params and you try to convert all of them into atoms?

File: apps/mintacoin_web/lib/mintacoin_web/views/accounts_view.ex [hunk 6]
```diff
@@ -0,0 +1,35 @@
+defmodule MintacoinWeb.AccountsView do
+  use MintacoinWeb, :view
+
+  alias Mintacoin.Account
+
+  @type template :: String.t()
+  @type assigns :: map()
+  @type json_template :: map()
+  @type account :: Account.t()
+  @type object :: map()
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/views/accounts_view.ex:10]: @miguelnietoa IMO, we should avoid using words that represent the OOP paradigm like `object`. What about we use `resource` instead?

File: apps/mintacoin_web/lib/mintacoin_web/views/accounts_view.ex [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  @type key :: atom()
+  @type value :: any()
+
+  @account_object %{object: "account"}
+  @public_attributes [:address, :email, :name, :derived_key, :seed_words, :signature, :status]
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/views/accounts_view.ex:15]: @miguelnietoa please, use a sigil instead

File: apps/mintacoin_web/test/mintacoin_web/controllers/accounts_controller_test.exs [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+  import Mintacoin.Factory
+
+  alias Mintacoin.{Minter, Encryption, Mnemonic, Keypair}
+
+  @create_attrs %{"name" => "Test Name", "email" => "test@mail.com"}
+  @update_attrs %{"name" => "Updated Test Name", "email" => "updatedtest@mail.com"}
+  @invalid_attrs %{}
+  @non_existing_address "78f0eb93-4466-451c-b3e7-5c6189500919"
+  @auth_header "authorization"
```
Comment (juanhurtado10) [inline apps/mintacoin_web/test/mintacoin_web/controllers/accounts_controller_test.exs:12]: @miguelnietoa Why are you using constants for the test assertions? IMO, it is not a good approach, the setup macro should be used instead

File: apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  def recover(conn, %{"address" => address, "seed_words" => seed_words}) do
+    case Accounts.recover_signature(address, seed_words) do
+      {:ok, signature} -> render(conn, "signature.json", signature: signature)
+      {:error, :not_found} -> {:error, :not_found}
+      {:error, :bad_argument} -> {:error, :bad_request}
+    end
+  end
+
+  @spec fetch_allowed_params(params :: params()) :: params()
+  defp fetch_allowed_params(params) do
```
Comment (juanhurtado10) [inline apps/mintacoin_web/lib/mintacoin_web/controllers/accounts_controller.ex:79]: @miguelnietoa this seems something that can be removed from the controller's logic. What about using a standalone module for these params transformations?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

**Comment by miguelnietoa** [comment]
Requested changes applied! We look forward to the new code review. 😄

**Pull request merged**